staking/tests/recover_account.ts

```diff
@@ -0,0 +1,260 @@
+        pythTokenMint: pythMintAccount.publicKey,
+        unlockingDuration: 2,
+        epochDuration: new BN(3600),
+        freeze: false,
+        pdaAuthority: pdaAuthority,
+        governanceProgram: governanceProgram,
+        pythTokenListTime: null,
+        agreementHash: getDummyAgreementHash(),
+        mockClockTime: new BN(10),
+      })
+      .rpc({
+        skipPreflight: DEBUG,
+      });
+
+    await program.methods
+      .createTarget(votingProduct)
+      .accounts({
+        targetAccount: votingProductMetadataAccount,
+        governanceSigner: provider.wallet.publicKey,
+      })
+      .rpc();
+
+    await requestPythAirdrop(
+      provider.wallet.publicKey,
+      pythMintAccount.publicKey,
+      pythMintAuthority,
+      PythBalance.fromString("100"),
+      provider.connection
+    );
+
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+  });
+
+  it("updates stake account owner with governance authority", async () => {
+    const governanceConnection = await StakeConnection.createStakeConnection(
+      program.provider.connection,
+      provider.wallet as NodeWallet,
+      program.programId
+    );
+
+    const newOwner = new Keypair();
+
+    const newOwnerAta = await Token.getAssociatedTokenAddress(
+      ASSOCIATED_TOKEN_PROGRAM_ID,
+      TOKEN_PROGRAM_ID,
+      pythMintAccount.publicKey,
+      newOwner.publicKey,
+      true
+    );
+
+    const createAtaIx = Token.createAssociatedTokenAccountInstruction(
+      ASSOCIATED_TOKEN_PROGRAM_ID,
+      TOKEN_PROGRAM_ID,
+      pythMintAccount.publicKey,
+      newOwnerAta,
+      newOwner.publicKey,
+      pythMintAuthority.publicKey
+    );
+
+    const instructions: TransactionInstruction[] = [createAtaIx];
+
+    const badStakeAccountAddress = await governanceConnection.withCreateAccount(
+      instructions,
+      newOwnerAta,
+      {
+        fullyVested: {},
+      }
+    );
+
+    const transaction: Transaction = new Transaction();
+    transaction.instructions.push(...instructions);
+
+    await governanceConnection.program.provider.sendAndConfirm(transaction, [
+      pythMintAuthority,
+    ]);
+
+    // The fix
+
+    const recoverAccountInstruction =
+      await governanceConnection.buildRecoverAccountInstruction(
+        badStakeAccountAddress,
+        provider.wallet.publicKey
+      );
+    const recoverAccountTransaction = new Transaction();
+    recoverAccountTransaction.instructions.push(recoverAccountInstruction);
+
+    await governanceConnection.program.provider.sendAndConfirm(
+      recoverAccountTransaction
+    );
+
+    const fixedAccount = await governanceConnection.loadStakeAccount(
+      badStakeAccountAddress
+    );
+
+    assert.equal(
+      fixedAccount.stakeAccountPositionsJs.owner.toString(),
+      newOwner.publicKey.toString()
+    );
+    assert.equal(
+      fixedAccount.stakeAccountMetadata.owner.toString(),
+      newOwner.publicKey.toString()
+    );
```

Comment:
Can we add these asserts before too, to assert the owner is newOwnerAta before we call the instruction

Reply:
Done